libs/akita/src/lib/arrayToggle.ts

```diff
@@ -0,0 +1,11 @@
+/**
+ * Adds or removes a value from an array by comparing its values. If a matching value exists it is removed, otherwise
+ * it is added to the array.
+ *
+ * @param array The array to modify.
+ * @param newValue The new value to toggle.
+ * @param compare A compare function to determine equality of array values.
+ */
+export function arrayToggle<T>(array: T[], newValue: T, compare: (a: T, b: T) => boolean = (a, b) => a === b) {
+  return !!~array.findIndex((oldValue) => compare(newValue, oldValue)) ? array.filter((oldValue) => !compare(newValue, oldValue)) : [...array, newValue];
```

Comment:
Out of curiosity :sweat_smile:, why not use `array.some` instead of a bitwise operation with `findIndex`? performance?

Reply:
Yes, for performance reasons. I usually prefer readability. I don't think it's such a difference.

Reply:
Okay, thanks :smile: